### PR TITLE
raft: add learner state for rebuildme

### PR DESCRIPTION
**Pull request by prettyshuang**

### CODE REVIEW

File: src/cli/cmd/mysql.go [hunk 1]
```diff
@@ -212,9 +212,9 @@ func mysqlRebuildMeCommandFn(cmd *cobra.Command, args []string) {
 
 	// 4. disable raft
```
Comment (TCeason) [inline src/cli/cmd/mysql.go]: Maybe  '4. disable raft, set backup state' is better?

File: src/cli/cmd/mysql.go [hunk 2]
```diff
@@ -328,11 +328,20 @@ func mysqlRebuildMeCommandFn(cmd *cobra.Command, args []string) {
 
 	// 16. enable raft
 	{
-		log.Warning("S16-->enable.raft.begin...")
-		if _, err := callx.EnableRaftRPC(self); err != nil {
```
Comment (TCeason) [inline src/cli/cmd/mysql.go]: Could add a test for invalid state rebuild.

**Comment by prettyshuang** [comment]
push update：
1. Rename TestRaftRPCHASetBackup to TestRaftRPCHASetBackupFromFollower
2. Add TestRaftRPCHASetBackupFromInvalid and TestRaftRPCHASetBackupFromIdle

**Comment by prettyshuang** [comment]
push update: BACKUP -> LEARNER

**Comment by BohuTANG** [comment]
LGTM

**Pull request merged**